code/modules/projectiles/guns/energy/special.dm

```diff
@@ -305,15 +305,140 @@
 
 /obj/item/gun/energy/toxgun
 	name = "plasma pistol"
-	desc = "A specialized firearm designed to fire lethal bolts of toxins."
+	desc = "A specialized firearm designed to fire heated bolts of plasma. Can be overloaded for a high damage shield breaking shot."
 	icon_state = "toxgun"
-	fire_sound = 'sound/effects/stealthoff.ogg'
-
+	item_state = "toxgun"
+	sprite_sheets_inhand = list("Vox" = 'icons/mob/clothing/species/vox/held.dmi', "Drask" = 'icons/mob/clothing/species/drask/held.dmi') //This apperently exists, and I have the sprites so sure.
 	w_class = WEIGHT_CLASS_NORMAL
 	origin_tech = "combat=4;magnets=4;powerstorage=3"
-	ammo_type = list(/obj/item/ammo_casing/energy/toxplasma)
+	ammo_type = list(/obj/item/ammo_casing/energy/toxplasma, /obj/item/ammo_casing/energy/charged_plasma)
 	shaded_charge = 1
 	can_holster = TRUE
+	atom_say_verb = "beeps"
+	bubble_icon = "swarmer"
+	light_color = "#89078E"
+	light_range = 3
+	light_power = 0.1
+	var/overloaded = FALSE
+	var/warned = FALSE
+	var/charging = FALSE
+
+/obj/item/gun/energy/toxgun/Initialize(mapload)
+	. = ..()
+	START_PROCESSING(SSfastprocess, src)
+
+/obj/item/gun/energy/toxgun/Destroy()
+	STOP_PROCESSING(SSfastprocess, src)
+	return ..()
+
+/obj/item/gun/energy/toxgun/process() //keep your finger on the trigger soldier
+	..()
+	if(overloaded)
+		cell.charge -= 0.5 //2.5 per second, 25 every 10 seconds
+		if(cell.charge <= 25 && !warned)
+			warned = TRUE
+			playsound(loc, 'sound/weapons/smg_empty_alarm.ogg', 75, 1)
+			atom_say("Caution, charge low. Forced discharge in under 10 seconds.")
+		if(cell.charge <= 5)
+			discharge()
+		if(ishuman(loc))
+			var/mob/living/carbon/human/user = loc
+			if(!user.is_in_active_hand(src))
+				discharge()
+		else
+			discharge()
+
+/obj/item/gun/energy/toxgun/attack_self(mob/living/user)
+	if(overloaded)
+		to_chat(user, "<span class='warning'>[src] is already overloaded!</span>")
+		return
+	if(cell.charge <= 140) //at least 6 seconds of charge time
+		to_chat(user, "<span class='warning'>[src] does not have enough charge to be overloaded.</span>")
+		return
+	to_chat(user, "<span class='notice'>You begin to overload [src].</span>")
+	charging = TRUE
+	if(do_after(user, 25, target = src))
+		overloaded = TRUE
+		cell.charge -= 125
+		playsound(loc, 'sound/machines/terminal_prompt_confirm.ogg', 75, 1)
+		atom_say("Overloading successful.")
+		light_power = 4 //extra visual effect to make it more noticable to user and victims alike
```

Comment:
use the `set_light` proc so this properly updates.

Reply:
Done